### DISCUSSION TITLE
Required Permissions?

### DISCUSSION THREAD

**Discussion by AlexisTDeRuyter** [fieldOfView/Cura-OctoPrintPlugin#263] [Q&A]
Octoprint encourages using Application Keys rather than the global API token and will be deprecating the global token in the future.  As such I would like to create a Cura user in Octoprint and follow the security principle of least privilege.  I've looked through the ReadMe but don't see any details on which specific permissions are required, just that an API token is needed.  At a bare minimum I imagine the below are needed, is anyone able to provide which, if any, other permissions are required?

- `File Upload`
- `Print`
- `Status`
- `Webcam`

I'm not able to find documentation on the available permissions in the Octoprint docs, but the core ones are detailed [in the code here starting on line 278](https://github.com/OctoPrint/OctoPrint/blob/master/src/octoprint/access/permissions.py) and the permissions can also be seen in the Octoprint GUI.

Thanks!

**Comment by fieldOfView** [comment]
The `Control` (used to jog and preheat in the monitor tab) and `Settings Access` (used to test the API key and to see if a webcam and sdcard printing etc is available) are also required. In the not-so-distant future, the `File List` permission may also be useful to allow the user to select from a list of available folders to save the gcode to.
